alg/teca_2d_component_area.cxx

```diff
@@ -285,6 +288,20 @@ const_p_teca_dataset teca_2d_component_area::execute(
 
     teca_metadata &out_metadata = out_mesh->get_metadata();
 
+    // get the background_id, and pass it through
+    long bg_id = this->background_id;
+    if (this->background_id < 0)
```

Comment:
It just occurred to me that this precludes datasets that have the background labeled as a negative value.   This is something that we might actually come across; for example if making use of the '_FillValue' attribute in netCDF, a person might 'remove' the background in a segmented image by setting it to a fill value; this fill value is commonly a negative number like -9999 or -1e-20.  I've actually done exactly this for segmented images before: set the background to -9999 and set the _FillValue attribute to -9999.

One idea--though maybe clunky--could be to have two properties/options:
```
bool override_background_id(false);
long background_id = 0;
```

This would then change the check from `if(this->background_id <0)` to `if ( ! this->override_background_id )`.

Reply:
I have made the change (will share shortly) to use -1 as a code for "get background_id form the metadata", and -2 as a code for "no background label". This preclude the use of -1 and -2 in mask array but would allow any other negative value to be used. How do you feel ab out that?

Reply:
I think that seems okay; I haven't seen any cases where someone might use -1 or -2 as the background label